doc/reference/multisig.md

```diff
@@ -8,57 +8,230 @@ The multi-signature scripts are written as JSON objects. We have three types of
+            "type": "requireSignature"
+        },
+        {
+            "paymentKeyHash": "3a94d6d4e786a3f5d439939cafc0536f6abc324fb8404084d6034bf8",
+            "type": "requireSignature"
+        },
+        {
+            "paymentKeyHash": "b12e094d1db7c0fba5121f22db193d0060efed8be43654f861bb68ae",
+            "type": "requireSignature"
+        },
+        {
+            "paymentKeyHash": "9be49d56442b4b8b16cab4e43e238bbdefc6c803d554c82fcd5facc3",
+            "type": "requireSignature"
+        },
+        {
+            "paymentKeyHash": "622be5fab3b5c3f371a50a535e4d3349c942a98cecee93b24e2fd11d",
+            "type": "requireSignature"
+        }
+    ],
+    "type": "any"
 }
 ```
 
 ## atLeast
 
-The "atLeast" key indicates that in order to spend this tx output, we require "atLeast" 2 corresponding signatures from the list of payment key hashes. The "required" key indicates the minimum number of signatures we need to spend the tx output.
+The `"type"` key's value, "atLeast", indicates that in order to spend this tx output, we require "atLeast" 2 corresponding signatures from the list of payment key hashes. The "required" key indicates the minimum number of signatures we need to spend the tx output.
 
 ```json
 {
-    "atLeast": {
-        "list": [
-            "2f3d4cf10d0471a1db9f2d2907de867968c27bca6272f062cd1c2413",
-            "f856c0c5839bab22673747d53f1ae9eed84afafb085f086e8e988614",
-            "b275b08c999097247f7c17e77007c7010cd19f20cc086ad99d398538",
-            "686024aecb5884d73a11b9ae4e63931112ba737e878d74638b78513a"
-        ],
-        "required": 2
-    }
+    "scripts": [
+        {
+            "paymentKeyHash": "2f3d4cf10d0471a1db9f2d2907de867968c27bca6272f062cd1c2413",
+            "type": "requireSignature"
+        },
+        {
+            "paymentKeyHash": "f856c0c5839bab22673747d53f1ae9eed84afafb085f086e8e988614",
+            "type": "requireSignature"
+        },
+        {
+            "paymentKeyHash": "b275b08c999097247f7c17e77007c7010cd19f20cc086ad99d398538",
+            "type": "requireSignature"
+        },
+        {
+            "paymentKeyHash": "686024aecb5884d73a11b9ae4e63931112ba737e878d74638b78513a",
+            "type": "requireSignature"
+        }
+    ],
+    "required": 2,
+    "type": "atLeast"
 }
 ```
+### Example of using multi-signature scripts
+
+Let's walk through how one would use a multi-signature script. This is a multistep step process, involving the creation of a multi-signature address, sending ADA to that address followed by gathering the required witnesses in order to spend the ADA from the multi-signature address. We will demonstrate this with an `all` script.
+
+#### Sending ada to a script address
+
+#### Step 1 - Create multi-signature script
+
+You will first need to generate the keys that you require witnessing from via the `cardano-cli shelley  address key-gen ` command. Then you construct the multi-signature script as follows:
```

Comment:
Remove preceding space after trailing back tick.

Remove extra space from the double space after `shelley`.